src/index.ts

```diff
@@ -114,54 +157,48 @@ export default class SwupScrollPlugin extends Plugin {
 
 	/**
 	 * Get an element based on anchor
-	 * @param {string} hash
-	 * @returns {mixed}
 	 */
-	getAnchorElement = (hash = '') => {
+	getAnchorElement = (hash: string = ''): Element | null => {
 		// Look for a custom function provided via the plugin options
 		if (typeof this.options.getAnchorElement === 'function') {
 			return this.options.getAnchorElement(hash);
-		} else {
-			return this.swup.getAnchorElement(hash);
 		}
+
+		return this.swup.getAnchorElement(hash);
 	};
 
 	/**
 	 * Get the offset for a scroll
-	 * @param {(HtmlELement|null)} element
-	 * @returns {number}
 	 */
-	getOffset = (element = null) => {
+	getOffset = (el?: Element): number => {
+		if (!el) return 0;
 		// If options.offset is a function, apply and return it
 		if (typeof this.options.offset === 'function') {
-			return parseInt(this.options.offset(element), 10);
+			return Math.round(this.options.offset(el));
 		}
 		// Otherwise, return the sanitized offset
-		return parseInt(this.options.offset, 10);
```

Comment:
The idea here was to allow unit values returned from CSS, e.g. `10px`. Does `Math.round` still support that?

Reply:
Aah, that's why it was there! No, lets keep to parseInt then.

Reply:
So "10em/rem/pt/vh" would also resolve to 10 px? 

Not ideal, but we can't change it without a major version. Will revert it to parseInt.

Reply:
Technically yes, but the use case is for return values of `window.getComputedStyle` which are always in `px`. Not sure why somebody would pass in a string like `10em/rem/pt/vh`.

Reply:
We can add a comment about expecting `window.getComputedStyle` return values here so we don't forget what this was for... Not sure if you can type that without going mad.

Reply:
We _could_ use a regex to extract the first group of number-like characters `[0-9\.]` and run `parseFloat` on that for extra safety. Would also be more accurate than `parseInt`. Maybe some other day :)